dataclassy/dataclass.py

```diff
@@ -51,16 +51,19 @@ def __new__(mcs, name, bases, dict_, **kwargs):
         # determine whether a post-initialisation method was defined by the user
         init = dict_.get('__init__')
         post_init = (type(init) is Function and not hasattr(init, '__dataclass__')
-                     or type(dict_.get('__post_init__')) is Function)
+                     or any(type(b.get('__post_init__')) is Function
```

Comment:
I think you could also do this by just moving the `post_init` check below where `all_defaults` is merged into `dict_`?

Reply:
It doesn't work.  I don't think everything from the superclass `dict_`s makes it into `all_defaults`

Reply:
You're right, specifically it gets overwritten later in the inheritance chain. I merged the check for `post_init` into the first loop over the bases which I'll push at some point.